### ISSUE TITLE
Packets

### ISSUE THREAD

**Issue by joery1337** [MasonBurdette/slither-client#1]
Are you going to document all the packets?

**Comment by MasonBurdette** [comment]
Yes.  That will probably be in the wiki.  I'm still figuring out everything.

**Comment by ItzLevvie** [comment]
Slither source code: http://slither.io/s/game36.js 
Use [Dan's tools](http://www.danstools.com/javascript-beautify/) to beautify the code above, to make it easier to read.

**Comment by MasonBurdette** [comment]
@ItzLevvie yes I already know that.  I made a function that `console.log()` whenever it does `ws.send()`

**Comment by Megabyte918** [comment]
Cant w8 till this is released :D

**Comment by firelightning13** [comment]
me 2 :D

**Comment by ThunderGemios10** [comment]
Yes! Once the packets are documented, I can finally start working on Clither! (Slither.io Server implementation)

**Comment by Megabyte918** [comment]
i love clither

**Comment by Coder-256** [comment]
@Megabyte918 Why is Clither any bether? It's still an empty shell (so far)

@ThunderGemios10 I think it would be best to create a server in JavaScript, so that Clither is compatible with slither-client. After all, the web client is written in Javascript (and the real server might be too). @MasonBurdette Maybe you could even expose an external API...

**Comment by ThunderGemios10** [comment]
@Coder-256 I have my reasons to choose Java, Also every project is an empty shell :P I can't start working without any info on Slither.io's packets..

**Comment by joery1337** [comment]
@Coder-256 because node.js is just bad.

**Comment by Coder-256** [comment]
@ThunderGemios10 @joery1337 Does anybody care to share reasons?

**Comment by joery1337** [comment]
From the experience I've had using node.js for Agar.io, it just sucked.

`I think it would be best to create a server in JavaScript, so that Clither is compatible with slither-client`
It does not matter what language you use, Slither.io uses WebSockets and that's not a JavaScript-only protocol.

`After all, the web client is written in Javascript (and the real server might be too`
I highly doubt this, with ~400 to 600 players per server, this would just not work out.

**Comment by Coder-256** [comment]
@joery1337 You still have not raised a good argument against node.js. If you have a valid reason (not one such as "it just sucked" or "this would just not work out"), then I would be glad to hear it. For my reasoning about Javascript (in general) vs Java, I think you misunderstood. What I meant is I thought it would be good if there was an API created by this project in Javascript and that API could be used by many different applications (which I think was @MasonBurdette's main goal).

One more reason I think node.js would be better than Java is that node.js is faster and handles memory better.

InfoWorld wrote an article that goes into the [pros and cons of each language](http://www.infoworld.com/article/2883328/java/java-vs-nodejs-an-epic-battle-for-developer-mindshare.html) in detail.

@ThunderGemios10 If you are still intent on using Java, then the best way to go is probably to use [Rhino](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/Rhino) along with this project's API (or, if you really want to, you could rewrite this project in Java which is what it looks like you were going to do).

**Comment by joery1337** [comment]
@Coder-256 Your arguments are also not valid, the only argument you gave is `is faster and handles memory`, you could say the same about any other language.

I'm not sure if you know where @ThunderGemios10 is talking about, it seems like you are talking about  a client, he is talking about a server. You are comparing slither-client (Client, Node.js) and Clither (Server, Java), it makes no sense.

**Comment by ThunderGemios10** [comment]
@joery1337 Uh yes, I'm only developing a server software for Slither.io, Nothing to do with client...

**Comment by Porama6400** [comment]
@Coder-256 
> I think it would be best to create a server in JavaScript, so that Clither is compatible with slither-client. After all, the web client is written in Javascript (and the real server might be too). 

No , JavaScript is a very unstable single thread language , it's not good for server

**Comment by ThunderGemios10** [comment]
You guys can find info on Packets here: https://github.com/ClitherProject/Slither.io-Protocol..

For now, It has info of Packet "a"

**Comment by MasonBurdette** [comment]
@ThunderGemios10 I am using Node.JS for the sole purpose to make it fast and easy to make headless clients.

**Comment by ThunderGemios10** [comment]
@MasonBurdette Node.js has only one thread and it is slow, But not very slow, Java however has multi-threading, Allowing performance to be fast.

**Comment by MasonBurdette** [comment]
@ThunderGemios10 if you use npm cluster you can multi-thread..  It's up to the programmers of the client to increase performance.

**Comment by MasonBurdette** [comment]
Anyone can read the readme.md to see that I am looking for a qualified person to take over..

**Comment by ThunderGemios10** [comment]
@MasonBurdette, That's fine, Someone is helping documenting the protocol too anyway, We have discovered 2 packets right now, Check https://github.com/ClitherProject/Slither.io-Protocol
